### DISCUSSION TITLE
Add retry logic for network calls in prone to error environments

### DISCUSSION THREAD

**Discussion by HeavyWombat** [paketo-buildpacks/feedback#9] [General]
I was wondering whether there would be an option to allow for a retry logic for certain network calls (i.e. downloads). The use case is that in our backend system the build of a Go app starts but fails with a `read tcp 194.31.147.142:40042->12.225.252.87:443: read: connection reset by peer` during *Executing build process*, which fails the whole build. So it would interesting to know if there could be an option to either retry an individual call within the build, or the build process itself automatically. Thanks in advance for your feedback.

**Comment by fg-j** [comment]
@HeavyWombat would it be possible to share a complete build log where this occurred? "Executing build process" is a pretty generic log line; I'm wondering which buildpacks/build processes are typically impacted by network flakes.

For what it's worth, I've definitely seen something similar when building Java apps that pull in a whole ton of dependencies.

**Reply by HeavyWombat** [reply]
Unfortunately, I only have some log output from our internal tooling that caught the issue, but not the full logs. I can share what we were able to see and what brought us to the question whether or not some kind of retry logic could help.

```
===> DETECTING
2 of 5 buildpacks participating
paketo-buildpacks/go-dist  0.3.0
paketo-buildpacks/go-build 0.2.2
===> ANALYZING
Previous image with name "icr.io/namespace/name:tag" not found
===> RESTORING
===> BUILDING
Paketo Go Distribution Buildpack 0.3.0
  Resolving Go version
    Candidate version sources (in priority order):
      <unknown> -> ""

    Selected Go version (using <unknown>): 1.15.8

  Executing build process
    Installing Go 1.15.8
read tcp 194.31.147.142:40042->12.225.252.87:443: read: connection reset by peer
ERROR: failed to build: exit status 1
```

For a bit more context, we use Paketo Buildpacks in https://github.com/shipwright-io/build, so this all runs in a Kube cluster.

**Reply by fg-j** [reply]
@ForestEckhardt Is this what it would look like if [postal](https://github.com/paketo-buildpacks/packit/tree/main/postal) failed to download the dependency?
